app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,27 @@ def show
     @user = User.find(params[:user_id])
     @post = @user.posts.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author = current_user
+
+    respond_to do |_format|
+      if @post.save
```

Comment:
**[OPTIONAL but highly recommended]** As mentioned by the previous :point_up: reviewer, in your code a post is created users should be redirected to the post, but that does not happen, the app remains on the new post page. You may want to look into that so kindly implement this as well :slightly_smiling_face: